src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -71,8 +83,14 @@ def _set_params(self, metric_names=None, qm_params=None, peak_sign=None, seed=No
             if "peak_sign" in qm_params_[k] and peak_sign is not None:
                 qm_params_[k]["peak_sign"] = peak_sign
 
+        try:
+            existing_metric_names = self.sorting_analyzer.get_extension("quality_metrics").params.get("metric_names")
+            metric_names_for_params = np.concatenate((existing_metric_names, metric_names))
+        except:
```

Comment:
I don't know if a naked except is the best move. This can hid bugs that become very difficult to track down. Is there a better exception to catch?

Reply:
Yeah, good point. There are possible exceptions: there can be no extension or the list of metric_names can be empty. So I've replaced this with two `if`s which should be more explicit. Whad'ya think?

Reply:
See my new comment below!